### PR TITLE
Issue-#5382: Revamp the turn icons used in Mapbox Android Nav SDK

### LINKED ISSUE #5382: Revamp the turn icons used in Mapbox Android Nav SDK
Mapbox Android Navigation SDK uses a default set of turn icons in the form of vector drawables. These vector drawables are used when showing maneuver turn instructions. However, some of these vector drawables are confusing and need to be revamped. There are customer [issues](https://github.com/mapbox/1tap-android/issues/1547) being reported as well in regards to incorrect icons. Issues with icons have been brought up verbally very often and we should take this up on a priority to fix them.

Vector drawables used are attached. 
[turnicons.zip](https://github.com/mapbox/mapbox-navigation-android/files/7900746/turnicons.zip)

Unfortunately I don't have the `.svg` files for these drawables.

/cc @mapbox/navigation-android @lawsonkight

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
### Description

Closes #5382 

#### Changes to modules

_libnavui-maneuver:_
- replaced all icons (except `mapbox_ic_depart*` and `mapbox_ic_roundabout*`) with maneuver vector icons from navigation-ui-resources
- updated strokeColor and fillColor of the new drawables to use ?attr/maneuverTurnIconColor and ?attr/maneuverTurnIconShadowColor references

_qa-test-app:_
- added IconsPreviewActivity for previewing all libnavui-maneuver module icons
- added custom IconPreview view
- added custom GenerateIconsPreviewTask gradle task that generates layout file with icon preview for each icon drawable in libnavui-maneuver/src/main/res/drawable folder

### Verification
On device:
1. Run `qa-test-app`
2. View "Icons Preview" activity

In Android Studio
1.  Open `qa-test-app/src/main/res/layout/layout_activity_icons_preview.xml`
2. Toggle "Split" view
3. Use mouse to scroll icons list

To alter icon colors, modify `Theme.IconsPreviewActivity` in `qa-test-app/src/main/res/values/themes.xml`

### Screenshots

| old icons | new icons |
|:----:|:----:|
| old_1 <br/> old_2 <br/> old_3 <br/> old_4 <br/> old_5 <br/> old_6 <br/> old_7 <br/> old_8 <br/> old_9 <br/> old_10 <br/> old_11 <br/> old_12 <br/> old_13 <br/> old_14 | new_1 <br/> new_2 <br/> new_3 <br/> new_4 <br/> new_5 <br/> new_6 <br/> new_7 <br/> new_8 <br/> new_9 <br/> new_10 <br/> new_11 <br/> new_12 <br/> new_13 <br/> new_14

### CODE REVIEW

**Comment by tomaszrybakiewicz** [comment]
Waiting for https://github.com/mapbox/mapbox-navigation-android/issues/5382#issuecomment-1026034258 before wrapping this up.

**Comment by tomaszrybakiewicz** [comment]
This one is good to go. Any additional icons we will add in a separate PR.

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_left.xml [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
-    android:viewportHeight="20"
-    android:viewportWidth="20"
-    android:width="20dp">
-  <path
-      android:fillColor="?attr/maneuverTurnIconColor"
-      android:pathData="M3,12a2,2,0,1,1,2-2a2,2,0,0,1-2,2Zm10.00293-0.96332a4.05782,4.05782,0,0,1,3.98877,4.07324h0.008300000000001972v1.37775a0.51232 0.51232,0,0,0,0.5123299999999986 0.5123299999999986h0.97534a0.51232 0.51232,0,0,0,0.5123300000000022-0.5123299999999986v-1.4876700000000014h-0.013850000000001472a6.05607,6.05607,0,0,0-5.9834-5.96332l-0.011-0.00183l0.00012 0.02008h-0.9918700000000005v-0.009090000000000487a0.35757 0.35757,0,0,1-0.37738-0.54242l1.34375-2.95184a0.38659 0.38659,0,0,0-0.62354-0.40942l-6.342829999999999,4.90368l6.34283,4.90369a0.3866 0.3866,0,0,0,0.62354-0.40942l-1.34375-2.95184a0.35757 0.35757,0,0,1,0.3773800000000005-0.5424299999999977v0.00909h1" />
-</vector>
\ No newline at end of file
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="arriveRight"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_left.xml:5]: arriveLeft

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_straight.xml [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
-    android:viewportHeight="20"
-    android:viewportWidth="20"
-    android:width="20dp">
-  <path
-      android:fillColor="?attr/maneuverTurnIconColor"
-      android:pathData="M10,5a2,2,0,1,1,2-2a2,2,0,0,1-2,2Zm4.91284,8.35114l-4.90368-6.342839999999999l-4.903689999999999,6.342839999999999a0.38659 0.38659,0,0,0,0.40942 0.62354l2.95184-1.34375a0.35542 0.35542,0,0,1,0.5409600000000001 0.3690700000000007h-0.007690000000000197v5.50006a0.49992 0.49992,0,0,0,0.4999400000000005 0.4999399999999987h1.00012a0.49992 0.49992,0,0,0,0.4999399999999987-0.4999399999999987v-5.491760000000001h0.00916a0.35757 0.35757,0,0,1,0.54242-0.37738l2.95184,1.34375a0.3866 0.3866,0,0,0,0.4094200000000008-0.6235300000000006Z" />
-</vector>
\ No newline at end of file
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="arrive"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_straight.xml:5]:
```suggestion
    android:name="arriveStraight"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_left.xml [hunk 3]
```diff
@@ -1,14 +1,39 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- fork (flipped) -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="fork"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_left.xml:5]:
```suggestion
    android:name="forkLeft"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_right.xml [hunk 4]
```diff
@@ -1,14 +1,38 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- fork -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="fork"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_right.xml:5]:
```suggestion
    android:name="forkRight"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_slight_left.xml [hunk 5]
```diff
@@ -1,14 +1,39 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- fork (flipped) -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="fork"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_slight_left.xml:5]:
```suggestion
    android:name="forkSlightLeft"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_slight_right.xml [hunk 6]
```diff
@@ -1,14 +1,38 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- fork -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="fork"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_slight_right.xml:5]:
```suggestion
    android:name="forkSlightRight"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_straight.xml [hunk 7]
```diff
@@ -1,17 +1,38 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- fork -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="fork"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_fork_straight.xml:5]:
```suggestion
    android:name="forkStraight"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_left.xml [hunk 8]
```diff
@@ -1,14 +1,39 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- merge (flipped) -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="merge"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_left.xml:5]:
```suggestion
    android:name="mergeLeft"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_right.xml [hunk 9]
```diff
@@ -1,14 +1,38 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- merge -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="merge"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_right.xml:5]:
```suggestion
    android:name="mergeRight"
```

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_slight_left.xml [hunk 10]
```diff
@@ -1,14 +1,39 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?><!-- Generated by PaintCode - http://www.paintcodeapp.com -->
+<!-- merge (flipped) -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="20dp"
-    android:height="20dp"
-    android:viewportWidth="20"
-    android:viewportHeight="20">
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:name="merge"
```
Comment (abhishek1508) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_slight_left.xml:5]:
```suggestion
    android:name="mergeSlightLeft"
```

Comment (tomaszrybakiewicz) [reply libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_slight_left.xml:5]: @abhishek1508 Are we using those names for anything?

Comment (tomaszrybakiewicz) [reply libnavui-maneuver/src/main/res/drawable/mapbox_ic_merge_slight_left.xml:5]: I've only seen `android:name` used on a `<group>` and `<path>` tags. Never in `<vector>`.  They were probably added by PaintCode during export.

**Comment by tomaszrybakiewicz** [comment]
@abhishek1508 I've rebased and fixed merge conflicts. Removed unused `android:name` attributes from `<vector>` tags of all new icon drawables.

**Review by abhishek1508 — APPROVED** [review]

**Pull request merged**